src/crypto_tools/rng.rs

```diff
@@ -59,11 +59,40 @@ pub(crate) fn rng_seed<K>(
     Ok(ChaCha20Rng::from_seed(seed))
 }
 
+/// Initialize a RNG by hashing the arguments.
+/// Intended for use generating a ECDSA signing key.
+pub(crate) fn rng_seed_ecdsa_signing_key(
+    tag: u8,
+    secret_recovery_key: &SecretRecoveryKey,
+    session_nonce: &[u8],
+) -> TofnResult<impl CryptoRng + RngCore> {
+    if session_nonce.len() < SESSION_NONCE_LENGTH_MIN
+        || session_nonce.len() > SESSION_NONCE_LENGTH_MAX
+    {
+        error!(
+            "invalid session_nonce length {} not in [{},{}]",
+            session_nonce.len(),
+            SESSION_NONCE_LENGTH_MIN,
+            SESSION_NONCE_LENGTH_MAX
+        );
+        return Err(TofnFatal);
+    }
+
+    let mut prf = Hmac::<Sha256>::new(secret_recovery_key.0[..].into());
+
+    prf.update(&tag.to_be_bytes());
+    prf.update(session_nonce);
+
+    let seed = prf.finalize().into_bytes().into();
```

Comment:
prf seems to have the `secret_recovery_key`. Does it need to be zeroized?

Reply:
Looks like `into()` is not causing a clone.  I traced the execution to [here](https://docs.rs/generic-array/0.14.4/src/generic_array/lib.rs.html#553-563).  This implementation of `From` converts one reference to another without copying.  Let's hope it stays that way. 😳 I added an explicit declaration and comments.